contracts/voting/dao-voting-native-staked/src/contract.rs

```diff
@@ -273,6 +404,73 @@ pub fn execute_claim(
         .add_attribute("amount", release))
 }
 
+pub fn execute_update_active_threshold(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    new_active_threshold: Option<ActiveThreshold>,
+) -> Result<Response, ContractError> {
+    let dao = DAO.load(deps.storage)?;
+    if info.sender != dao {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    if let Some(active_threshold) = new_active_threshold {
+        match active_threshold {
+            ActiveThreshold::Percentage { percent } => {
+                if percent > Decimal::percent(100) || percent.is_zero() {
+                    return Err(ContractError::InvalidActivePercentage {});
+                }
+            }
+            ActiveThreshold::AbsoluteCount { count } => {
+                let denom = DENOM.load(deps.storage)?;
+                assert_valid_absolute_count_threshold(deps.as_ref(), &denom, count)?;
+            }
+        }
+        ACTIVE_THRESHOLD.save(deps.storage, &active_threshold)?;
+    } else {
+        ACTIVE_THRESHOLD.remove(deps.storage);
+    }
+
+    Ok(Response::new().add_attribute("action", "update_active_threshold"))
+}
+
+pub fn execute_add_hook(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    addr: String,
+) -> Result<Response, ContractError> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    if Some(info.sender.clone()) != config.owner && Some(info.sender.clone()) != config.manager {
```

Comment:
perhaps worth adding a check on instantiation that owner and manager are not the same

Reply:
IMO, going to replace this outdated pattern with `cw-ownable` like we are using elsewhere.

Reply:
Also, IMO not really a problem if they are the same. Have an existing ticket for the `cw-ownable` work. May pass on it in my follow on PR because there is already a lot there.